src/hooks/useAsync.js

```diff
@@ -0,0 +1,25 @@
+import { useCallback, useState } from "react";
+
+function useAsync(asyncFunction) {
+  const [pending, setPending] = useState(false);
+  const [error, setError] = useState(null);
+
+  const wrappedFunction = useCallback(
+    async (...args) => {
+      setPending(true);
+      setError(null);
+      try {
+        return await asyncFunction(...args);
+      } catch (error) {
+        setError(error);
+      } finally {
+        setPending(false);
+      }
+    },
+    [asyncFunction]
+  );
+
+  return [pending, error, wrappedFunction];
```

Comment:
```suggestion
  return {pending, error, wrappedFunction};
```
이런 경우는 배열 보단 객체를 사용하는게 좋아 보여요. 그래야 호출하는쪽에서 순서가 아닌 이름으로 데이터들을 사용할 수 있기 때문이에요.